pkg/metrics/metric/collections/product.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package collections
+
+func CartesianProduct[T any](vs ...[]T) VecIterator[T] {
```

Comment:
Not sure why we are using the interface here vs the actual type?

nit: Docs?

Reply:
Since we probably don't actually need to worry about memory access patterns since this is no longer a general library, I've simplified this down to just take and return a 2d vector [][]T.